kafka-ui-react-app/src/components/Schemas/Edit/Edit.tsx

```diff
@@ -0,0 +1,183 @@
+import { useForm } from 'react-hook-form';
+import { isEqual } from 'lodash';
+
+export interface EditProps {
+  subject: SchemaName;
+  schema: SchemaSubject;
+  clusterName: ClusterName;
+  schemasAreFetched: boolean;
+  createSchema: (
+    clusterName: ClusterName,
+    newSchemaSubject: NewSchemaSubject
+  ) => Promise<void>;
+  fetchSchemasByClusterName: (clusterName: ClusterName) => void;
+  updateSchemaCompatibilityLevel: (
+    clusterName: ClusterName,
+    subject: string,
+    compatibilityLevel: CompatibilityLevelCompatibilityEnum
+  ) => void;
+}
+
+export default function Edit({
+  subject,
+  schema,
+  clusterName,
+  schemasAreFetched,
+  createSchema,
+  fetchSchemasByClusterName,
+  updateSchemaCompatibilityLevel,
+}: EditProps) {
+  React.useEffect(() => {
+    if (!schemasAreFetched) fetchSchemasByClusterName(clusterName);
+  }, [clusterName, fetchSchemasByClusterName]);
+
+  let newSchema = '';
+  let isSubmitting = false;
+  const { register, handleSubmit } = useForm<NewSchemaSubjectRaw>();
+  const handleSchemaChange = (e: string) => {
+    newSchema = e;
+  };
+  const history = useHistory();
+  const onSubmit = async ({
+    schemaType,
+    compatibilityLevel,
+  }: {
+    schemaType: SchemaType;
+    compatibilityLevel: CompatibilityLevelCompatibilityEnum;
+  }) => {
+    try {
+      isSubmitting = true;
+      if (
+        (newSchema &&
+          !isEqual(JSON.parse(schema.schema), JSON.parse(newSchema))) ||
+        schemaType !== schema.schemaType
+      ) {
+        await createSchema(clusterName, {
+          ...schema,
+          schema: newSchema || schema.schema,
+          schemaType: schemaType || schema.schemaType,
+        });
+      }
+      if (compatibilityLevel !== schema.compatibilityLevel) {
+        await updateSchemaCompatibilityLevel(
+          clusterName,
+          subject,
+          compatibilityLevel
+        );
+      }
+      history.push(clusterSchemaPath(clusterName, subject));
+    } catch (e) {
+      isSubmitting = false;
+      // Show Error
+    }
+  };
+
+  return (
+    <div className="section">
+      <div className="level">
+        <div className="level-item level-left">
+          <Breadcrumb
+            links={[
+              {
+                href: clusterSchemasPath(clusterName),
+                label: 'Schema Registry',
+              },
+              {
+                href: clusterSchemaPath(clusterName, subject),
+                label: subject,
+              },
+            ]}
+          >
+            Edit
+          </Breadcrumb>
+        </div>
+      </div>
+
+      {schemasAreFetched && !isSubmitting ? (
+        <div className="box">
+          <div className="columns">
+            <div className="column is-one-half">
+              <h4 className="title is-5 mb-2">Latest Schema</h4>
+              <JSONEditor
+                readonly
+                value={JSON.stringify(JSON.parse(schema.schema), null, '\t')}
```

Comment:
pls memo it somewhere

Reply:
I can't use useMemo, because it is evaluated right away, but the computation requires the value of `schema`, which is not there until it is loaded from the backend. So at the moment useMemo is evaluated, `schema` is undefined. That's why I have to use useCallback, which will be evaluated only after the schemas are loaded.